
Generations:   0%|                                                                                                                                             | 0/20 [00:00<?, ?it/s]/Users/leo/Programming/PLR/Leo/gen_algo_opt_folder/gen_algo2.py:140: RuntimeWarning: invalid value encountered in double_scalars
  probabilities = [score / total_fitness for score in scores]
Generations:   0%|                                                                                                                                             | 0/20 [01:14<?, ?it/s]
Traceback (most recent call last):
  File "/Users/leo/Programming/PLR/Leo/gen_algo_opt_folder/gen_algo2.py", line 195, in <module>
    best_individual, best_fitnesses, variances, best_columns, n_features = ga.run()
                                                                           ^^^^^^^^
  File "/Users/leo/Programming/PLR/Leo/gen_algo_opt_folder/gen_algo2.py", line 141, in run
    parents_indices = np.random.choice(len(population), size=int(self.population_size * self.selection_rate), p=probabilities, replace=False)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "mtrand.pyx", line 935, in numpy.random.mtrand.RandomState.choice
ValueError: probabilities contain NaN
Generation 1/20: Best Fitness = 0.9127017855644226, Best Genome = (1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 1, 0, 1, 1, 1, 0, 1, 0, 0, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 1, 0, 1, 0, 0, 0.7022176197850185, 0.04139025188927134, 29, 50)